addons/account/models/account_report.py

```diff
@@ -364,7 +361,16 @@ def _create_report_expression(self, engine):
                 'formula': formula.lstrip(' \t\n'),  # Avoid IndentationError in evals
                 'subformula': subformula
             }
-            vals_list.append(vals)
+            if report_line.expression_ids:
+                # expressions already exists, update the first expression with the right engine
+                # since syntactic sugar aren't meant to be used with multiple expressions
+                for expression in report_line.expression_ids:
+                    if expression.engine == engine:
+                        expression.write(vals)
+                        continue
```

Comment:
break, not continue, otherwise you're updating all the expressions with this engine